gse-util/src/main/java/com/powsybl/gse/util/NodeChooser.java

```diff
@@ -445,9 +445,16 @@ private void setDragOverStyle(TreeTableCell<N, N> treeTableCell) {
         }
     }
 
+    private boolean isChildOf(TreeItem<N> targetTreeItem) {
+        return targetTreeItem == dragAndDropMove.getSourceTreeItem().getParent();
+    }
+
+    public boolean isMovable(N item, TreeItem<N> targetTreeItem) {
+        return dragAndDropMove != null && !isChildOf(targetTreeItem);
+    }
 
-    private void dragOverEvent(DragEvent event, Object item, TreeTableRow<N> treeTableRow, TreeTableCell<N, N> treeTableCell) {
-        if (item instanceof Folder && item != dragAndDropMove.getSource()) {
+    private void dragOverEvent(DragEvent event, N item, TreeTableRow<N> treeTableRow, TreeTableCell<N, N> treeTableCell) {
+        if (item instanceof Folder && item != dragAndDropMove.getSource() && isMovable(item, treeTableRow.getTreeItem())) {
```

Comment:
add a condition checking if dragAndDropMove is null like in ProjectPane to avoid to throw errors if it is the case:
```if (item instanceof Folder && dragAndDropMove != null && ...)```

Reply:
item != dragAndDropMove.getSource() and !isChildOf(targetTreeItem) don't do the same thing ? 
-the first one forbid  a dnd to itself
- the second one forbid to drag a dile to it's own parent